### PR TITLE
Fix #5236 [2.8][Translation] specify additional translation loading paths

### LINKED ISSUE #5236: [2.8][Translation] specify additional translation loading paths
This allow to add custom resources path via prepend config or in config.yml

```yml
framework:
    translator:
        paths:
            - %kernel.root_dir%/../vendor/Symfony/Component/Form/Resources/translations
```

see https://github.com/symfony/symfony/pull/14561

### PR DESCRIPTION
**Pull request by Balamung**
My humble try of contributing from the SymfonyCon Hackday ! :)

| Q | A |
| --- | --- |
| Doc fix? | no |
| New docs? | yes |
| Applies to | 2.8 |
| Fixed tickets | #5236 |

### CODE REVIEW

File: book/translation.rst [hunk 1]
```diff
@@ -374,6 +374,18 @@ For more options, see :ref:`component-translator-message-catalogs`.
 
 .. note::
 
+    The paths at which Symfony will look for message files can be modified by adding
+    your custom resources path in your config.yml file:
```
Comment (WouterJ) [inline book/translation.rst]: I think we can make this more direct and shorter (shorter = better is quite an important rule in documentations): ``You can add other translation paths with the ``paths`` option in the configuration:``

File: book/translation.rst [hunk 2]
```diff
@@ -374,6 +374,18 @@ For more options, see :ref:`component-translator-message-catalogs`.
...
+
+    .. code-block:: yaml
+
+        framework:
+            translator:
+                paths:
+                    - %kernel.root_dir%/../vendor/Symfony/Component/Form/Resources/translations
```
Comment (WouterJ) [inline book/translation.rst]: You shoudl document the other formats as well:
```rst
.. note::

    You can add other directories with the ``paths`` option in the configuration:

    .. configuration-block::

        .. code-block:: yaml

            # app/config/config.yml
            framework:
                translator:
                    paths:
                        - "%kernel.root_dir%/../translations"

        .. code-block:: xml

            <?xml version="1.0" encoding="UTF-8" ?>
            <container xmlns="http://symfony.com/schema/dic/services"
                xmlns:framework="http://symfony.com/schema/dic/symfony"
                xmlns:xsi="http://www.w3.org/2001/XMLSchema-Instance"
                xsi:schemaLocation="http://symfony.com/schema/dic/services
                    http://symfony.com/schema/dic/services/services-1.0.xsd
                    http://symfony.com/schema/dic/symfony
                    http://symfony.com/schema/dic/symfony/symfony-1.0.xsd"
            >

                <framework:config>
                    <framework:translator>
                        <framework:path>%kernel.root_dir%/../translations</framework:path>
                    </framework:translator>
                </framework:config>
            </container>

        .. code-block:: php

            $container->loadFromExtension('framework', array(
                'translator' => array(
                    'paths' => array(
                        '%kernel.root_dir%/../translations'
                    )
                ),
            ));
```

**Comment by WouterJ** [comment]
Thank you @Balamung! I added one more comment (you're responding a bit too fast :wink: ). After that, this PR is ready to be merged imo.

**Comment by Balamung** [comment]
Wow, thank you so much for providing the whole code ! I'll get sure to integrate this in the PR right away!

File: .idea/dictionaries/pmaraitre.xml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+<component name="ProjectDictionaryState">
```
Comment (WouterJ) [inline .idea/dictionaries/pmaraitre.xml]: you commited a backup file from your editor, please remove it from your commits

Comment (Balamung) [inline .idea/dictionaries/pmaraitre.xml]: So sorry about that, I feel dumb for messing up things like this :(

**Comment by WouterJ** [comment]
:+1:

File: .idea/dictionaries/pmaraitre.xml (hunk 3 above)
Comment (WouterJ) [inline .idea/dictionaries/pmaraitre.xml]: Don't worry, you're fast in fixing them :)

Btw, a little tip if you're going to create more PRs: Create a new branch for each change you make. This way, each PR has its own branch and changes in PR 1 aren't included in PR 2.

You can create new branches like this:
```
# move to the 2.8 branch
$ git checkout 2.8
# create a new branch
$ git checkout -b my_awesome_doc_fix
# ... work on the changes
# commit
$ git add path/to/changes.rst
$ git commit -m "I did something great!'
# push to github & create a PR
$ git push ...
```

**Comment by xabbuh** [comment]
Thanks @Balamung for this great pull request. I have merged it and made some minor tweaks in 2ad1ba5ca1ccf222e6bab65e72c6e462d546d3dd afterwards.

**Comment by xabbuh** [comment]
Oh and congratulations for your first contribution. :)

**Comment by Balamung** [comment]
Well thanks ! But again, thanks @WouterJ Who did almost all the work ! =)

**Pull request merged**